### DISCUSSION TITLE
Enable generating typescript ABIs for multiple Foundry deploy scripts

### DISCUSSION THREAD

**Discussion by MattPereira** [scaffold-eth/scaffold-eth-2#896] [Ideas]
The current foundry version of SE-2 is only capable of exporting deployment info for a single deploy script because it depends on [this line](https://github.com/scaffold-eth/scaffold-eth-2/blob/e1b39d098bd3e0bb5096aecb3cac3020553138cb/packages/foundry/scripts-js/generateTsAbis.js#L82) where a specific file name is declared

Additionally, [this logic](https://github.com/scaffold-eth/scaffold-eth-2/blob/e1b39d098bd3e0bb5096aecb3cac3020553138cb/packages/foundry/scripts-js/generateTsAbis.js#L139-L148) overwrites the entire contents of the `nextjs/contracts/deployedContracts.ts` file so a viable solution would have to involve appending deployed contracts to that file but only if the deployment comes from a different script, otherwise you would want to overwrite the contract name keys because it would be newer version of same contracts ( if that makes sense :sweat_smile: )

For example, I'm hoping to be able to `exportDeployments()` from all of the [numbered deploy scripts for scaffold balancer](https://github.com/balancer/scaffold-balancer-v3/tree/main/packages/foundry/script). The solution of deploying all the pool factories in the [`00_DeploySetup.s.sol`](https://github.com/balancer/scaffold-balancer-v3/blob/ccad4ab282a63d491133e540b8a40c5e42ff15a5/packages/foundry/script/00_DeploySetup.s.sol#L30-L32) is viable for now, but will grow increasingly messy as more example pool and hooks contracts are added to the build. Would prefer to be able to encapsulate all the contracts related to eachother in each script while maintaining all the scaffold integrations on the frontend because I'm listening for pool created events emitted by the factory contract in order to offer buttons for the [pool explorer page](https://scaffold-balancerv3.vercel.app/pools)

**Comment by carletex** [comment]
Hey @MattPereira 

Could you check if what Shiv did for the create-eth foundry template could work? Maybe we can take some ideas from there and put it on SE-2 (where foundry is going to be the default)

**Reply by carletex** [reply]
Sorry, forgot the link to the PR: https://github.com/scaffold-eth/create-eth/pull/43

**Reply by MattPereira** [reply]
Absolutely, the general pattern Shiv implemented for the CLI extension DX definitely works :pray: 

@technophile-04 also elaborates on how this could be integrated for the default SE-2 foundry repo [with this comment](https://github.com/scaffold-eth/scaffold-eth-2/pull/781#issuecomment-2018642266) :raised_hands: 

Much appreciated @carletex! So glad to be part of this amazing community where y'all are always two steps ahead :heart:

**Reply by technophile-04** [reply]
>For example, I'm hoping to be able to exportDeployments() from all of the [numbered deploy scripts for scaffold balancer](https://github.com/balancer/scaffold-balancer-v3/tree/main/packages/foundry/script)

Yup even I had similar thing my mind but there doesn't seem a good/famous library in foundry world to handle this properly although there were discussion regarding it https://github.com/foundry-rs/foundry/issues/4732 but seems stale now :/ , also added some notes in  https://github.com/scaffold-eth/scaffold-eth-2/pull/898
